pkg/assets/create/create_test.go

```diff
@@ -0,0 +1,269 @@
+package create
+
+import (
+	"bytes"
+	"context"
+	"os"
+	"strings"
+	"sync"
+	"testing"
+	"time"
+
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/discovery"
+	"k8s.io/client-go/dynamic"
+	dynamicfake "k8s.io/client-go/dynamic/fake"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/restmapper"
+	ktesting "k8s.io/client-go/testing"
+
+	"github.com/openshift/library-go/pkg/assets"
+)
+
+func init() {
+	fetchLatestDiscoveryInfoFn = func(dc *discovery.DiscoveryClient) (meta.RESTMapper, error) {
```

Comment:
why not a var?

Reply:
it is a var?

Reply:
(also I need mutex to prevent data race when testing discovery refresh in-flight)

Reply:
oh, we are in test here.